src/metamorpho/MetamorphoSnippets.sol

```diff
@@ -103,28 +130,50 @@ contract MetamorphoSnippets {
 
     // // --- MANAGING FUNCTIONS ---
 
-    // deposit in the vault a nb of asset
-    function depositInVault(uint256 assets, address onBehalf) public returns (uint256 shares) {
-        shares = vault.deposit(assets, onBehalf);
+    /// @notice Deposit `assets` into the `vault` on behalf of `onBehalf`.
+    /// @dev Sender must approve the snippets contract to manage his tokens before the call.
+    /// @param vault The address of the metamorpho vault.
+    /// @param assets the amount to deposit.
+    /// @param onBehalf The address that will own the increased deposit position.
+    function depositInVault(address vault, uint256 assets, address onBehalf) public returns (uint256 shares) {
+        ERC20(IMetaMorpho(vault).asset()).transferFrom(msg.sender, address(this), assets);
+
+        _approveMaxVault(vault);
+
+        shares = IMetaMorpho(vault).deposit(assets, onBehalf);
+    }
+
+    /// @notice Withdraws `assets` from the `vault` on behalf of the sender, and sends them to `receiver`.
+    /// @dev Sender must approve the snippets contract to manage his tokens before the call.
+    /// @param vault The address of the metamorpho vault.
+    /// @param assets the amount to withdraw.
+    /// @param receiver The address that will receive the withdrawn assets.
+    function withdrawFromVaultAmount(address vault, uint256 assets, address receiver)
+        public
+        returns (uint256 redeemed)
+    {
+        redeemed = IMetaMorpho(vault).withdraw(assets, receiver, msg.sender);
     }
 
-    // withdraw from the vault a nb of asset
-    function withdrawFromVaultAmount(uint256 assets, address onBehalf) public returns (uint256 redeemed) {
-        address receiver = onBehalf;
-        redeemed = vault.withdraw(assets, receiver, onBehalf);
+    /// @notice Withdraws the whole sender's position from the `vault`, and sends the withdrawn amount to `receiver`.
+    /// @param vault The address of the metamorpho vault.
+    /// @param receiver The address that will receive the withdrawn assets.
+    function withdrawFromVaultAll(address vault, address receiver) public returns (uint256 redeemed) {
+        uint256 assets = IMetaMorpho(vault).maxWithdraw(msg.sender);
+        redeemed = IMetaMorpho(vault).withdraw(assets, receiver, msg.sender);
     }
```

Comment:
Do we want to promote this usecase? I would rather promote using `redeem` with the user's balance to withdraw all

Reply:
I agree, should we delete this function to only expose `redeemAllFromVault` ?